### ISSUE TITLE
[1.5.0-M1] Scala 2.13-3.x sandwich does not work with Scala.JS

### ISSUE THREAD

**Issue by xuwei-k** [sbt/sbt#6280]
## steps

```diff
diff --git a/Main.scala b/Main.scala
new file mode 100644
index 0000000..9effcd7
--- /dev/null
+++ b/Main.scala
@@ -0,0 +1,7 @@
+package example
+
+object Main {
+  def main(args: Array[String]): Unit = {
+    println(scalaz.Maybe.just(2))
+  }
+}
diff --git a/build.sbt b/build.sbt
new file mode 100644
index 0000000..4fc7548
--- /dev/null
+++ b/build.sbt
@@ -0,0 +1,7 @@
+enablePlugins(ScalaJSPlugin)
+
+scalaVersion := "3.0.0-M3"
+
+scalaJSUseMainModuleInitializer := true
+
+libraryDependencies += "org.scalaz" %%% "scalaz-core" % "7.3.3" cross CrossVersion.for3Use2_13
diff --git a/project/build.properties b/project/build.properties
new file mode 100644
index 0000000..b5a4bf1
--- /dev/null
+++ b/project/build.properties
@@ -0,0 +1 @@
+sbt.version=1.5.0-M1
diff --git a/project/plugins.sbt b/project/plugins.sbt
new file mode 100644
index 0000000..904909d
--- /dev/null
+++ b/project/plugins.sbt
@@ -0,0 +1 @@
+addSbtPlugin("org.scala-js" % "sbt-scalajs" % "1.4.0")

```

execute `sbt "print fullClasspath" run`

- https://github.com/xuwei-k/sbt-1.5.0-M1-sandwich-scala-js/commit/a0d34060f565354934fcbe5f6d9648b296256283
- https://github.com/xuwei-k/sbt-1.5.0-M1-sandwich-scala-js/runs/1754256620?check_suite_focus=true

## problem

```
* Attributed(/home/runner/work/sbt-1.5.0-M1-sandwith-js-native/sbt-1.5.0-M1-sandwith-js-native/target/scala-3.0.0-M3/classes)
* Attributed(/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-library_sjs1_3.0.0-M3/3.0.0-M3/scala3-library_sjs1_3.0.0-M3-3.0.0-M3.jar)
* Attributed(/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-js/scalajs-library_2.13/1.4.0/scalajs-library_2.13-1.4.0.jar)
* Attributed(/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scalaz/scalaz-core_2.13/7.3.3/scalaz-core_2.13-7.3.3.jar)
* Attributed(/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.4/scala-library-2.13.4.jar)
[info] Fast optimizing /home/runner/work/sbt-1.5.0-M1-sandwith-js-native/sbt-1.5.0-M1-sandwith-js-native/target/scala-3.0.0-M3/sbt-1-5-0-m1-sandwith-js-native-fastopt
[error] Referring to non-existent class scalaz.Maybe$
[error]   called from example.Main$.main([java.lang.String)void
[error]   called from static example.Main.main([java.lang.String)void
[error]   called from core module module initializers
[error] involving instantiated classes:
[error]   example.Main$
[error] Referring to non-existent class scalaz.Maybe$
[error]   called from example.Main$.main([java.lang.String)void
[error]   called from static example.Main.main([java.lang.String)void
[error]   called from core module module initializers
[error] involving instantiated classes:
[error]   example.Main$
[error] Referring to non-existent method scalaz.Maybe$.just(java.lang.Object)scalaz.Maybe
[error]   called from example.Main$.main([java.lang.String)void
[error]   called from static example.Main.main([java.lang.String)void
[error]   called from core module module initializers
[error] involving instantiated classes:
[error]   example.Main$
[error] Referring to non-existent class scalaz.Maybe$
[error]   called from example.Main$.main([java.lang.String)void
[error]   called from static example.Main.main([java.lang.String)void
[error]   called from core module module initializers
[error] involving instantiated classes:
[error]   example.Main$
[error] Cannot access module for non-module scalaz.Maybe$
[error]   called from example.Main$.main([java.lang.String)void
[error]   called from static example.Main.main([java.lang.String)void
[error]   called from core module module initializers
[error] involving instantiated classes:
[error]   example.Main$
[error] Referring to non-existent class scalaz.Maybe
[error]   called from example.Main$.main([java.lang.String)void
[error]   called from static example.Main.main([java.lang.String)void
[error]   called from core module module initializers
[error] involving instantiated classes:
[error]   example.Main$
[error] There were linking errors
[error] (Compile / fastLinkJS) There were linking errors
```

## expectation

`"org.scalaz" %%% "scalaz-core" % "7.3.3" cross CrossVersion.for3Use2_13` should reference `scalaz-core_sjs1_2.13/7.3.3/scalaz-core_sjs1_2.13-7.3.3.jar` not `scalaz/scalaz-core_2.13/7.3.3/scalaz-core_2.13-7.3.3.jar`

## notes

`sbt-dotty 0.5.1`, `sbt 1.4.6`  with `withDottyCompat scalaVersion.value` works

- https://github.com/sbt/sbt/pull/5984
- https://github.com/sbt/sbt/issues/6080

**Comment by xuwei-k** [comment]
🤔 

https://github.com/sbt/librarymanagement/blob/7b45fc594f7122741bdf0805420ce259ce0eefb3/core/src/test/scala/sbt/librarymanagement/CrossVersionTest.scala#L285

**Comment by eed3si9n** [comment]
https://github.com/sbt/librarymanagement/blob/v1.5.0-M4/core/src/main/scala/sbt/librarymanagement/CrossVersion.scala#L45-L51
`sbt.librarymanagement.Binary` has prefix and suffix fields.

`%%%` implementation https://github.com/portable-scala/sbt-platform-deps/blob/v1.0.0/src/main/scala/org/portablescala/sbtplatformdeps/PlatformDepsBuilders.scala#L36-L43 seems to set these prefix, and DottyPlugin's [`withDottyCompat`](https://github.com/lampepfl/dotty/blob/sbt-dotty-v0.5.1/sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala#L126-L127) preserves the prefix and suffixes from the existing Binary value:

```scala
              if (compatVersion.nonEmpty)
                moduleID.cross(CrossVersion.constant(binary.prefix + compatVersion + binary.suffix))
```

In other words, `withDottyCompat` _composes_, but in its current form `.cross(CrossVersion.for3Use2_13)` does not.

This is an oversight on my part that I did not see when I promoted the use of `.cross(...)`.

## workaround

Maybe a workaround might be to change the type of `CrossVersion.for3Use2_13` so we can implement an overload of `.cross(...)` method that doesn't just overwrite the existing value.

**Comment by eed3si9n** [comment]
Here's my PR for this - https://github.com/sbt/librarymanagement/pull/364
